SIPS/sip-26.md

```diff
@@ -0,0 +1,168 @@
+---
+sip: 26
+title: Account Router Snaps
+status: Draft
+author: Daniel Rocha (@danroc)
+created: 2024-08-28
+---
+
+## Abstract
+
+This SIP proposes a new API to be implemented by a new Account Router, allowing
+the forwarding of signing requests to the appropriate account Snap (i.e., Snaps
+that implement the [Keyring API][keyring-api]).
+
+## Motivation
+
+The Keyring API is being modified to support non-EVM chains. However, a
+challenge arises in identifying the correct account Snap that should receive
+the signing request, as this information is often only obtainable from the
+request itself, which varies based on the method and chain.
+
+## Specification
+
+> Formal specifications are written in TypeScript.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### High-Level architecture
+
+The diagram below represents a high-level architecture of how the Account
+Router integrates with the RPC Router, Account Snaps, and Protocol Snaps.
+
+![High-level architecture](../assets/sip-26/components-diagram.png)
+
+- **Account Snaps**: Snaps that implement the Keyring API and are responsible
+  for signing requests and managing accounts.
+
+- **Protocol Snaps**: Snaps that implement protocol methods that do not require
+  an account to be executed.
+
+- **RPC Router**: Native component that forwards non-signing requests to the
+  appropriate Protocol Snap or native implementation.
+
+- **Account Router**: Native component that forwards signing requests to the
+  appropriate Account Snap or native implementation.
+
+- **Account Address Resolution Snaps**: Snaps that implement the
+  `resolveAccountAddress` method to extract the account address from the
+  request object.
+
+### Account Router
+
+The Account Router exposes the [`snap_manageAccounts`][snap-manage-accs] method
+to allow account Snaps to register, remove, and update accounts.
+
+For example, this code can be used by an Account Snap to register a new account
+with the Account Router:
+
+```typescript
+// This will notify the Account Router that a new account was created, and the
+// Account Router will register this account as available for signing requests
+// using the `eth_signTypedData_v4` method.
+await snap.request({
+  method: 'snap_manageAccounts',
+  params: {
+    method: 'notify:accountCreated',
+    params: {
+      account: {
+          id: '74bb3393-f267-48ee-855a-2ba575291ab0',
+          type: 'eip155:eoa',
+          address: '0x1234567890123456789012345678901234567890',
+          methods: ['eth_signTypedData_v4'],
+          options: {},
```

Comment:
what would go in `options`?

Reply:
This is part of the Keyring API and is used by Snap devs to store implementation-dependent information about the account. An example would be to use it to store the tresholds, index, and other metadata used by MPC accounts.